File: Defensoria-back/Dev2/primercontacto/src/main/java/ipn/escom/defensoria/primercontacto/entity/CitaPrimerContacto.java
package ipn.escom.defensoria.primercontacto.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Entity
@Table(name = "citas_primer_contacto")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CitaPrimerContacto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "queja_id", nullable = false)
    private Long quejaId;

    @Column(name = "folio", nullable = false, length = 50)
    private String folio;

    @Column(name = "quejoso_id")
    private Long quejosoId;

    @Column(name = "quejoso_nombre", length = 150)
    private String quejosoNombre;

    @Column(name = "analista_id", nullable = false)
    private Long analistaId;

    @Column(name = "analista_nombre", length = 150)
    private String analistaNombre;

    @Column(name = "fecha_cita", nullable = false)
    private LocalDate fechaCita;

    @Column(name = "hora_cita", nullable = false)
    private LocalTime horaCita;

    @Column(name = "tipo_cita", nullable = false, length = 50)
    private String tipoCita;

    @Column(name = "motivo", nullable = false, columnDefinition = "TEXT")
    private String motivo;

    @Column(name = "estatus", nullable = false, length = 50)
    private String estatus;

    @Column(name = "fecha_creacion", nullable = false)
    private LocalDateTime fechaCreacion;
}

File: Frontend/PrimerContacto/frontend-primer-contacto/src/app/core/models/cita-primer-contacto.ts
export interface CitaPrimerContacto {
  id?: number;
  expedienteId?: number;
  folio: string;

  quejosoId?: number;
  quejoso: string;

  analistaId?: number;
  analistaNombre?: string;

  fecha: string;
  hora: string;

  tipo: 'Presencial' | 'Virtual';
  motivo: string;
  estatus: string;

  fechaCreacion?: string;
}

export interface CrearCitaPrimerContacto {
  folio: string;

  quejosoId?: number;
  quejosoNombre?: string;

  analistaId: number;
  analistaNombre: string;

  fechaCita: string;
  horaCita: string;
  tipoCita: string;
  motivo: string;
}